### ISSUE TITLE
Linear algebra resources

### ISSUE THREAD

**Issue by hanjiexi** [ossu/computer-science#423]
In issue #418, a number of students raised issues about the course we're currently recommending for linear algebra ([LAFF](https://www.edx.org/course/linear-algebra-foundations-frontiers-utaustinx-ut-5-05x-0)). The main complaint seems to be that students start getting lost pretty quickly because challenging concepts are explained perhaps too theoretically and not enough in plain English.

A few alternatives, some of which were suggested by students:
- [Khan Academy Linear Algebra](https://www.khanacademy.org/math/linear-algebra)
- [Essence of Linear Algebra](https://www.youtube.com/playlist?list=PLZHQObOWTQDPD3MizzM2xVFitgF8hE_ab) video series
- Gilbert Strang's [book](https://www.amazon.com/Introduction-Linear-Algebra-Gilbert-Strang/dp/0980232775/)
- Gilbert Strang's [course](https://ocw.mit.edu/courses/mathematics/18-06-linear-algebra-spring-2010/) which is already on our [extras/courses](https://github.com/open-source-society/computer-science/blob/master/extras/courses.md) list

I'm open to replacing LAFF with Gilbert Strang's course, but I worry that Strang's course could be just as difficult to understand as LAFF without some introductory material. Khan Academy is also very compelling since it includes assignments. I would like to solicit more feedback from people who have experimented with these different resources to see what the best one is — which may end up being some combination of resources. Note also that a resource doesn't have to be "all-or-nothing" — we can say "the first five weeks of course X and then course Y", similar to what we did in [Intro CS](https://github.com/open-source-society/computer-science#introduction-to-computer-science).

Part of what makes LAFF so attractive is the integration of programming assignments. These should be considered mandatory for anyone learning linear algebra for CS. So my inclination is to keep LAFF in the curriculum, but add a short supplement before it, like Essence of Linear Algebra.

**Comment by DevStorage** [comment]
OSSU isт't a regular university, so you need to be more interactive and find information by yourself(because you interested in it, isn't you?). There are also a lot of people that don't want to learn and find information and we don't need to take them seriously.

**Comment by Heasummn** [comment]
I was the one that recommended adding LAFF into the OSSU curriculum because I felt the course at the time was not in-depth enough and could use some work. I do agree that LAFF also has some issues. I like the idea of recommending a starter course and then recommending LAFF. Giving the participant an introduction to Linear Algebra before throwing them into LAFF seems like something that would make the experience much easier.

**Comment by waciumawanjohi** [comment]
I second the suggestion to add Essence of Linear Algebra (EoAL) as an introduction and then LAFF after that. I was actually going to make a pull request to start a new section in Extras for videos (right now there are just Courses and Readings, neither of which fits Essence of Linear Algebra well). I think putting EoAL directly into the curriculum is an even better step.

Reasoning:

- Essence of Linear Algebra is an incredible resource. It does an even better job than Strang's materials at giving a high level view of what is going on. It does not offer the depth that students need, so it needs to be paired with another course.

- OSS should always prioritize resources that provide students with feedback. Feedback is one of the [7 Principles of Undergraduate Education](https://scholar.google.com/scholar?q=good+practice+in+undergraduate+education) (Chickering and Gamson, 1987), a work cited thousands of times. OSS can't yet assure all of the principles, but we can focus on courses with assignments that are graded and which give immediate feedback. That focus will aid all learners.

- Prioritization of feedback tells us to choose LAFF over Strang's course.

- I hope that MIT releases an MITx version of Linear Algebra in the near future. I think the Strang lectures focus on comprehension. I'm concerned that LAFF focuses overly much on efficient computation.

- @DevStorage if we are unwilling to consider why students have difficulty with the curriculum and how we can address that difficulty, then we don't need to spend any time to create a curriculum at all. If students trust OSS enough to follow the recommended curriculum, OSS needs to live up to that trust by continually evaluating where the curriculum falls short and how to improve it.

**Comment by tururay** [comment]
I don't know EoLA but Strang's material helped me a lot while studying LA. I always look for his material.

**Comment by raialok** [comment]
I found the book [Linear Algebra: Step by Step] (https://www.amazon.com/Linear-Algebra-Step-Kuldeep-Singh/dp/0199654441/ref=sr_1_4?ie=UTF8&qid=1505029993&sr=8-4&keywords=linear+algebra) very useful. The book starts from the basics and explains all concepts in detail. It also has complete solutions available on [publisher's website.] (http://global.oup.com/booksites/content/9780199654444/) The amazon reviews seem to confirm my opinion.

**Comment by shailalias** [comment]
I agree with @waciumawanjohi about the importance of feedback in a course. LAFF provides good information and valuable feedback, even if it is a little heavy at the start. Additional resources such as Khan Academy or lighter reading material as a background are always useful though.  That said, those extra resources should remain extra and not be included as part of the curriculum. However, it wouldn't hurt if they were suggested as an option prerequisite before starting LAFF.

**Comment by a1h05** [comment]
I have almost complete calculus one, and won't be possible without links provided in curriculum, because lectures sometimes do not provide enough information to solve their quizzes (after working through textbook it is feasible task). If LAFF has the same issue it would be better to provide link where students can find material which will help to successfully complete that course.

**Comment by hanjiexi** [comment]
I completely agree about the importance of feedback which is why I want to keep LAFF in the curriculum. It sounds like there is no major disagreement with keeping it there either.

So I tentatively plan to add Essence of Linear Algebra into the next curriculum update. It's come highly recommended from several sources, and it seems to bridge an important conceptual gap, so I feel comfortable making it part of the curriculum proper.
